python/cudf/cudf/core/column/column.py

```diff
@@ -1357,6 +1329,33 @@ def as_column(arbitrary, nan_as_null=None, dtype=None, length=None):
             elif np.issubdtype(data.dtype, np.datetime64):
                 data = data.fillna(np.datetime64("NaT"))
 
+    elif hasattr(arbitrary, "__array_interface__"):
+        # CUDF assumes values are always contiguous
+        desc = arbitrary.__array_interface__
+        shape = desc["shape"]
+        arb_dtype = np.dtype(desc["typestr"])
+        # CUDF assumes values are always contiguous
+        if len(shape) > 1:
+            raise ValueError("Data must be 1-dimensional")
+
+        arbitrary = np.asarray(arbitrary)
+        if not arbitrary.flags["C_CONTIGUOUS"]:
+            arbitrary = np.ascontiguousarray(arbitrary)
+
+        if dtype is not None:
+            arbitrary = arbitrary.astype(dtype)
+
+        if arb_dtype.kind == "M":
+            data = as_column(pa.array(arbitrary), nan_as_null=nan_as_null)
```

Comment:
Instead of going through a `pa.array` now, we should be able to create a `Buffer` from arbitrary and then do `as_column(buffer, dtype=arbitrary.dtype)` followed by the below `time_col_replace_nulls`.